pinot-core/src/test/java/org/apache/pinot/queries/RangeQueriesTest.java

```diff
@@ -223,6 +223,42 @@ public void testSelectionOverRangeFilter(String query, int min, int max, boolean
     }
   }
 
+  @Test(dataProvider = "selectionTestCases")
+  public void testSelectionOverRangeFilterAfterReload(String query, int min, int max, boolean inclusive)
```

Comment:
Query execution tests seem to be focused on range index only. Any reason ?

Reply:
We should add other tests like

- Run query on a raw column using that in filter
- Enable dict + reload
- Run same query again and it should use dict based predicate evaluator and return same result

Same goes for using the column in SELECT clause. It should correctly return the same result after doing dual lookup in rewriten fwd index and dict as it did before with raw fwd index. Things like that

Reply:
Couple of types for both SV and MV should be considered